scripts/setup-roles.ts

```diff
@@ -0,0 +1,89 @@
+    const signers: SignerWithAddress[] = await ethers.getSigners();
+    const [admin, borrower] = signers;
+    const deployer = admin;
+
+    if (!LOAN_CORE_ADDRESS) {
+        throw new Error("Must specify LOAN_CORE_ADDRESS in environment!");
+    }
+    if (!admin.address) {
+        throw new Error("Must specify ADMIN_ADDRESS in environment!");
+    }
+    if (FEE_CONTROLLER_ADDRESS) {
+        console.log(`Fee controller address: ${FEE_CONTROLLER_ADDRESS}`);
+    }
+
+    const loanCore = await ethers.getContractAt("LoanCore", `${LOAN_CORE_ADDRESS}`);
+    const lenderNote = await ethers.getContractAt("PromissoryNote", `${LENDER_NOTE_ADDRESS}`);
+    const borrowerNote = await ethers.getContractAt("PromissoryNote", `${BORROWER_NOTE_ADDRESS}`);
+    const vaultFactory = await ethers.getContractAt("VaultFactory", `${VAULT_FACTORY_ADDRESS}`);
+
+    //Grant correct permissions for promissory note
+    //Giving to user to call PromissoryNote functions directly
+    for (const note of [borrowerNote, lenderNote]) {
+        await note.connect(await admin).initialize(loanCore.address);
+    }
+
+    // grant the borrower: FEE_CLAIMER_ROLE
+    const updateBorrowerPermissions = await loanCore.connect(admin).grantRole(FEE_CLAIMER_ROLE, borrower.address)
+    await updateBorrowerPermissions.wait();
+
+    // set LoanCore admin and fee claimer
+    const updateLoanCoreFeeClaimer = await loanCore.connect(borrower).grantRole(FEE_CLAIMER_ROLE, ADMIN_ADDRESS);
+    await updateLoanCoreFeeClaimer.wait();
+
+    // grant LoanCore the admin role to enable authorizeUpgrade onlyRole(DEFAULT_ADMIN_ROLE)
+    const updateLoanCoreAdmin = await loanCore.grantRole(ADMIN_ROLE, ADMIN_ADDRESS);
+    await updateLoanCoreAdmin.wait();
+
+    // grant VaultFactory the admin role to enable authorizeUpgrade onlyRole(DEFAULT_ADMIN_ROLE)
+    const updateVaultFactoryAdmin = await vaultFactory.grantRole(ADMIN_ROLE, ADMIN_ADDRESS);
+    await updateVaultFactoryAdmin.wait();
+
+    // grant originationContoller the owner role to enable authorizeUpgrade onlyOwner
+    const updateOriginationControllerAdmin = await loanCore.grantRole(ORIGINATOR_ROLE, `${ORIGINATION_CONTROLLER_ADDRESS}`);
+    await updateOriginationControllerAdmin.wait();
+
+    // renounce ownership from deployer
+    const renounceAdmin = await loanCore.renounceRole(ADMIN_ROLE, await deployer.getAddress());
+    await renounceAdmin.wait();
+    // renounce ability to claim fees
+    const renounceFeeClaimer = await loanCore.renounceRole(FEE_CLAIMER_ROLE, await deployer.getAddress());
+    await renounceFeeClaimer.wait();
+
+    if (FEE_CONTROLLER_ADDRESS) {
+        // set FeeController admin
+        const feeController = await ethers.getContractAt("FeeController", FEE_CONTROLLER_ADDRESS);
+        const updateFeeControllerAdmin = await feeController.transferOwnership(admin.address);
+        await updateFeeControllerAdmin.wait();
+    }
+
+    console.log("Transferred all ownership.\n");
+}
```

Comment:
Aren't we missing an ownership transfer for `CallWhitelist` as well? Should follow the same `Ownable` pattern as `FeeController` (as opposed to the `AccessControl` pattern of `LoanCore`, `VaultFactory`, and `OriginationController`).

Reply:
Also doesn't `PunkRouter` also need an ownership transfer?